rules/xcodeproj.bzl

```diff
@@ -0,0 +1,263 @@
+    BAZEL_INSTALLER: "$BAZEL_STUBS_DIR/{installer_path}"
+    CC: "$BAZEL_STUBS_DIR/clang-stub"
+    CXX: $CC
+    CLANG_ANALYZER_EXEC: $CC
+    CODE_SIGNING_ALLOWED: false
+    DONT_RUN_SWIFT_STDLIB_TOOL: true
+    LD: "$BAZEL_STUBS_DIR/ld-stub"
+    LIBTOOL: /usr/bin/true
+    SWIFT_EXEC: "$BAZEL_STUBS_DIR/swiftc-stub"
+    SWIFT_OBJC_INTERFACE_HEADER_NAME: ""
+    SWIFT_VERSION: 5
+""".format(name = paths.split_extension(project_name)[0], bazel_workspace_root = script_dot_dots, bazel_path = ctx.attr.bazel_path, installer_path = ctx.executable.installer.short_path)
+
+    targets = []
+    if ctx.attr.include_transitive_targets:
+        targets = depset(transitive = _get_transitive(ctx.attr.deps, _TargetInfo, "targets")).to_list()
+    else:
+        targets = [t for t in _get_transitive(ctx.attr.deps, _TargetInfo, "target") if t]
+
+    if targets:
+        yaml += """\
+targets:
+"""
+        for target in targets:
+            yaml += """\
+    {name}:
+        sources: [{sources}]
+        type: {product_type}
+        platform: iOS
+        settings:
+            PRODUCT_NAME: {name}
+            BAZEL_PACKAGE: {package}
+            MACH_O_TYPE: {macho_type}
+        preBuildScripts:
+        - name: Build with bazel
+          script: |
+            set -eux
+            cd $BAZEL_WORKSPACE_ROOT
+
+            $BAZEL_PATH build $BAZEL_PACKAGE:{bazel_name}
+            $BAZEL_INSTALLER
+""".format(
+                name = target.name,
+                sources = ", ".join(['{path: "%s", group: "%s", validate: false}' % (paths.join(src_dot_dots, s.short_path), paths.dirname(s.short_path)) for s in target.srcs.to_list()]),
+                package = target.package,
+                bazel_name = target.bazel_name,
+                product_type = target.product_type,
+                macho_type = "staticlib" if target.product_type == "framework" else "$(inherited)",
+            )
+
+        yaml += "schemes:\n"
+        for target in targets:
+            if target.product_type == "framework":
+                continue
+            action = "test"
+            if target.product_type == "application":
+                action = "run"
+            yaml += """\
+    {name}:
+        build:
+            parallelizeBuild: false
+            buildImplicitDependencies: false
+            targets:
+                {name}: [{action}]
+        {action}:
+            targets:
+                - {name}
+""".format(name = target.name, action = action)
+    ctx.actions.write(xcodegen_yaml, yaml)
+
+    ctx.actions.run(
+        executable = ctx.executable._xcodegen,
+        arguments = ["--quiet", "--no-env", "--spec", xcodegen_yaml.path, "--project", project.dirname],
+        inputs = depset([xcodegen_yaml], transitive = [target.srcs for target in targets]),
+        outputs = [project],
+    )
+    install_script = ctx.actions.declare_file(
+        "%s-install-xcodeproj.sh" % ctx.attr.name,
+    )
+
+    install_script_sh = """set -eu
```

Comment:
Should we use templates for this script here? Seems easier to maintain...

Reply:
I added a template.